src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private string _name;
+
+        private const string WindowsName = "windows";
+        private const string LinuxName = "linux";
+        private const string OSXName = "osx";
+
+        private static readonly OSName s_windows = new OSName(WindowsName);
+        private static readonly OSName s_linux = new OSName(LinuxName);
+        private static readonly OSName s_osx = new OSName(OSXName);
+
+        public OSName(string name)
+        {
+            if (name.Equals("unix", StringComparison.OrdinalIgnoreCase))
+                _name = LinuxName;
+            else if (name.Equals("mac", StringComparison.OrdinalIgnoreCase) ||
+                name.Equals("macosx", StringComparison.OrdinalIgnoreCase))
```

Comment:
I made the name value private, since this is user-provided and does not have any specifications on its values, and i dont see the value in basing any logic on this string. So exposing it to RuntimeInfo, and changing codeflow based on this string in other calling apis, leads to many possibilities to be considered based on this value. I still prefer to keep this in OSName, is there a reason why this is bad?